File: OneCodeTeam/Get-MAC-of-local-Ethernet-and-Wireless80211-17238/[VB]-Get MAC of local Ethernet and Wireless80211 (VBMACAddress)/VB/VBMACAddress/WMIHelper/WMIClassAttribute.vb
'************************** Module Header *******************************'
' Module Name:  WMIPropertyAttribute.vb
' Project:      VBMACAddress
' Copyright (c) Microsoft Corporation.
'
' This attribute indicates that the class is related to a WMI object.
' 
' 
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'*************************************************************************'


Namespace WMIHelper

    <AttributeUsage(AttributeTargets.Class)>
    Public Class WMIClassAttribute
        Inherits Attribute

        ''' <summary>
        ''' The related WMI class name.
        ''' </summary>
        Public Property ClassName() As String

    End Class
End Namespace

File: OneCodeTeam/Get-MAC-of-local-Ethernet-and-Wireless80211-17237/[C#]-Get MAC of local Ethernet and Wireless80211 (CSMACAddress)/C#/CSMACAddress/WMIHelper/WMIClassAttribute.cs
/****************************** Module Header ******************************\
 * Module Name:  WMIPropertyAttribute.cs
 * Project:      CSMACAddress
 * Copyright (c) Microsoft Corporation.
 *
 * This attribute indicates that the class is related to a WMI object.
 * 
 * 
 * 
 * This source is subject to the Microsoft Public License.
 * See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
 * All other rights reserved.
 * 
 * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
 * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
 * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System;

namespace CSMACAddress.WMIHelper
{

    [AttributeUsage(AttributeTargets.Class)]
    public class WMIClassAttribute : Attribute
    {

        /// <summary>
        /// The related WMI class name.
        /// </summary>
        public string ClassName { get; set; }

    }
}
